server/hotrod/src/main/scala/org/infinispan/server/hotrod/OperationStatus.scala

```diff
@@ -33,7 +31,7 @@ object OperationStatus extends Enumeration {
 
    def withCompatibility(st: OperationStatus, isCompatibilityEnabled: Boolean): OperationStatus = {
       if (isCompatibilityEnabled) {
-         (st: @switch) match {
+         st match {
```

Comment:
AFAIK no if we use `enum` (which is IMHO what we want here), see e.g. [here](http://docs.oracle.com/javase/specs/jvms/se7/html/jvms-3.html#jvms-3.10):

> The Java Virtual Machine's tableswitch and lookupswitch instructions operate only on int data. Because operations on byte, char, or short values are internally promoted to int, a switch whose expression evaluates to one of those types is compiled as though it evaluated to type int.